site/content/in-dev/unreleased/configuration.md

```diff
@@ -0,0 +1,127 @@
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+#
+title: Configuring Apache Polaris (Incubating)
+linkTitle: Configuring Polaris
+type: docs
+weight: 550
+---
+
+## Overview
+
+This page provides information on how to configure Apache Polaris (Incubating). Unless stated
+otherwise, this information is valid both for Polaris Docker images (and Kubernetes deployments) as
+well as for Polaris binary distributions.
+
+> Note: for Production tips and best practices, refer to [Configuring Polaris for Production]({{% ref "configuring-polaris-for-production.md" %}}).
+
+First off, Polaris server runs on Quarkus, and uses its configuration mechanisms. Read Quarkus
+[configuration guide](https://quarkus.io/guides/config) to get familiar with the basics.
+
+Quarkus aggregates configuration properties from multiple sources, applying them in a specific order
+of precedence. When a property is defined in multiple sources, the value from the source with the
+higher priority overrides those from lower-priority sources.
+
+The sources are listed below, from highest to lowest priority:
+
+1. System properties: Properties set via the Java command line using `-Dproperty.name=value`.
+2. Environment variables.
+3. Settings in `$PWD/config/application.properties` file.
+4. The `application.properties` files packaged in Polaris.
+5. Default values: hardcoded defaults within the application.
+
+When using environment variables, the environment variable is converted to a property name using a
+special naming convention; e.g. `POLARIS_REALM_CONTEXT_REALMS` would match the
+`polaris.realm-context.realms` property. The environment variable name is derived from the property
+name by replacing all `.` with `_` and converting the name to upper case. See
+[here](https://smallrye.io/smallrye-config/Main/config/environment-variables/) for more details.
+
+Note: Kubernetes deployments also accept environment variables in lowercase and with dashes and
+dots. For example, the following env variables are equivalent:
+
+```yaml
+env:
+  - name: POLARIS_REALM_CONTEXT_REALMS
+    value: "realm1,realm2"
+  - name: polaris.realm-context.realms
+    value: "realm1,realm2"
+```
+
+As stated above, a configuration file can also be provided at runtime; it should be available
+(mounted) at `$PWD/config/application.properties` for Polaris server to recognize it. In Polaris
+official Docker images, this location is `/deployment/config/application.properties`.
+
+For Kubernetes deployments, the configuration file is typically defined as a `ConfigMap`, then
+mounted in the container at `/deployment/config/application.properties`. It can be mounted in
+read-only mode, as Polaris only reads the configuration file once, at startup.
+
+## Polaris Configuration Options Reference
+
+The following configuration options are available for Polaris:
+
+TODO
```

Comment:
Note: this requires running `./gradlew :polaris-config-docs-site:generateDocs` and link the generated docs here. I suggest that we tackle this separately.